docs/guide/obs-integrations.asciidoc

```diff
@@ -0,0 +1,179 @@
+guide will walk you through the setup process.
+
+Because logging frameworks and formats vary greatly between different programming languages,
+there is no one-size-fits-all approach for ingesting your logs into Elasticsearch.
+The following tips should hopefully get you going in the right direction:
+
+**Download Filebeat**
+
+There are many ways to download and get started with Filebeat.
+Read the {filebeat-ref}/filebeat-installation.html[Filebeat Installation] guide to determine which is best for you.
+
+**Configure Filebeat**
+
+Modify the {filebeat-ref}/filebeat-configuration.html[`filebeat.yml`] configuration file to your needs.
+Here are some recommendations:
+
+* Set `filebeat.inputs` to point to the source of your logs
+* Point Filebeat to the same Elastic Stack that is receiving your APM data
+  * If you're using Elastic cloud, set `cloud.id` and `cloud.auth`.
+  * If your using a manual setup, use `output.elasticsearch.hosts`.
+
+[source,yml]
+----
+filebeat.inputs:
+- type: log <1>
+  paths: <2>
+    - /var/log/*.log
+cloud.id: "staging:dXMtZWFzdC0xLmF3cy5mb3VuZC5pbyRjZWMNjN2Q3YTllOTYyNTc0Mw==" <3>
+cloud.auth: "elastic:YOUR_PASSWORD" <4>
+----
+<1> Configures the `log` input
+<2> Path(s) that must be crawled to fetch the log lines
+<3> Used to resolve the Elasticsearch and Kibana URLs for Elastic Cloud
+<4> Authorization token for Elastic Cloud
+
+**JSON logs**
+
+For JSON logs you can use the {filebeat-ref}/filebeat-input-log.html[`log` input] to read lines from log files.
+Here's what a sample configuration might look like:
+
+[source,yml]
+----
+filebeat.inputs:
+  json.keys_under_root: true <1>
+  json.add_error_key: true <2>
+  json.message_key: message <3>
+----
+<1> `true` copies JSON keys to the top level in the output document
+<2> Tells Filebeat to add an `error.message` and `error.type: json` key in case of JSON unmarshalling errors
+<3> Specifies the JSON key on which to apply line filtering and multiline settings
+
+**Parsing unstructured logs**
+
+Consider the following log that is decorated with the `transaction.id` and `trace.id` fields:
+
+[source,log]
+----
+2019-09-18 21:29:49,525 - django.server - ERROR - "GET / HTTP/1.1" 500 27 | elasticapm transaction.id=fcfbbe447b9b6b5a trace.id=f965f4cc5b59bdc62ae349004eece70c span.id=None
+----
+
+All that's needed now is an {filebeat-ref}/configuring-ingest-node.html[ingest node processor] to pre-process your logs and
+extract these structured fields before they are indexed in Elasticsearch.
+To do this, you'd need to create a pipeline that uses Elasticsearch's {ref}/grok-processor.html[Grok Processor].
+Here's an example:
+
+[source, json]
+----
+PUT _ingest/pipeline/log-correlation
+{
+  "description" : "...",
+  "processors": [
+    {
+      "grok": {
+        "field": "message", <1>
+        "patterns": ["%{GREEDYDATA:msg} | elasticapm transaction.id=%{DATA:transaction.id} trace.id=%{DATA:trace.id} span.id=%{DATA:span.id}"] <2>
```

Comment:
Maybe use the [`message`](https://www.elastic.co/guide/en/ecs/current/ecs-base.html) key as opposed to `msg` in order to be ECS-compliant? Should we also document how to parse the [`@timestamp`](https://www.elastic.co/guide/en/ecs/current/ecs-base.html) and [`log.level`](https://www.elastic.co/guide/en/ecs/current/ecs-log.html) out of the log?

The original (unparsed log) can be stored under [`log.original`](https://www.elastic.co/guide/en/ecs/current/ecs-log.html).

But if we do that it gets more and more complicated for the user. Maybe we should just give a basic example here and link to the ECS documentation in case users want to extract more structure out of the logs.

Reply:
I'm hesitant to get too complicated here, as there's already a lot of complexity in the basic log correlation setup. I like the idea of linking to the ECS docs